test/OpenTelemetry.Instrumentation.AspNet.Tests/HttpInListenerTests.cs

```diff
@@ -219,50 +219,50 @@ public void Dispose()
 
             Assert.Equal(
                 "Ok",
-                span.Tags.FirstOrDefault(i => i.Key == SpanAttributeConstants.StatusCodeKey).Value);
+                span.TagObjects.FirstOrDefault(i => i.Key == SpanAttributeConstants.StatusCodeKey).Value);
 
             Assert.Equal(
                 "OK",
-                span.Tags.FirstOrDefault(i => i.Key == SpanAttributeConstants.StatusDescriptionKey).Value);
+                span.TagObjects.FirstOrDefault(i => i.Key == SpanAttributeConstants.StatusDescriptionKey).Value);
 
             var expectedUri = new Uri(url);
-            var actualUrl = span.Tags.FirstOrDefault(i => i.Key == SemanticConventions.AttributeHttpUrl).Value;
+            var actualUrl = span.TagObjects.FirstOrDefault(i => i.Key == SemanticConventions.AttributeHttpUrl).Value;
 
             Assert.Equal(expectedUri.ToString(), actualUrl);
 
             // Url strips 80 or 443 if the scheme matches.
             if ((expectedUri.Port == 80 && expectedUri.Scheme == "http") || (expectedUri.Port == 443 && expectedUri.Scheme == "https"))
             {
-                Assert.DoesNotContain($":{expectedUri.Port}", actualUrl);
+                Assert.DoesNotContain($":{expectedUri.Port}", actualUrl as string);
             }
             else
             {
-                Assert.Contains($":{expectedUri.Port}", actualUrl);
+                Assert.Contains($":{expectedUri.Port}", actualUrl as string);
             }
 
             // Host includes port if it isn't 80 or 443.
             if (expectedUri.Port == 80 || expectedUri.Port == 443)
             {
                 Assert.Equal(
                     expectedUri.Host,
-                    span.Tags.FirstOrDefault(i => i.Key == SemanticConventions.AttributeHttpHost).Value as string);
+                    span.TagObjects.FirstOrDefault(i => i.Key == SemanticConventions.AttributeHttpHost).Value as string);
```

Comment:
I think we probably want to replace all the `TagObjects.FirstOrDefault` with `GetTagValue` introduced by @CodeBlanch in #1221?
Another non-blocking thing - we probably need to validate if there are duplicated entries (e.g. two `http.host` tags) and fail the test accordingly.

Reply:
if we are using SetTag everywhere we should not have 2 entries, right?

Reply:
Yes, and the test cases should enforce it.

Reply:
For example:
```csharp
span.AddTag("t", "foo");
span.AddTag("t", "bar");
span.SetTag("t", "baz");
```

What would be the expected behavior? Currently I don't know, and I think we need to have some clarity.

Reply:
Right now, when we do that, it replaces the first "t" with "baz". So we will have 2 "t" with "baz" and "bar".
I will update those tests to check how many with that tag we have to be sure.

Reply:
For the `GetTagValue`, I saw that that is in `OpenTelemetry.Api`. Should I add to compile in tests as well to use it everywhere?

Reply:
Let's see what would @cijothomas suggest.

Reply:
@eddynaka I checked a few test projects, they all seem to be referencing SDK (which references API) or API directly. That means it should already be available. You might have to add `using OpenTelemetry.Trace`.

Reply:
@eddynaka Check out the commit I just pushed. Works fine! I was even able to remove the OpenTelemetry SDK ref, because the projects were doing Instrumentation -> SDK -> API. Why does this work? It has been a while since I read up on this, but I think the answer is: Because the unit test projects are SDK-style, project references flow like package references. Meaning, transitive dependencies become references. If these were traditional project types, I think it would work like you are saying and we would need direct references?

I can't find great link, only: [RestoreProjectStyle](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#using-packagereference-for-a-project-with-no-packagereferences). My understanding is SDK-style projects do that by default. If you use that switch in the old style, they will too.

Reply:
I see. Didn't know about that. I will take a look at the link. Thank you for the change. I was thinking the extension would be from tag object instead of the activity itself. My bad.